containers/Ideas/__tests__/IdeaProject.test.js

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import '@testing-library/jest-native/extend-expect';
+import { render, fireEvent, act } from '@testing-library/react-native';
+import { IdeaProject } from '../IdeaProject';
+import API from '../../../BaseApi';
+
+const promise = Promise.resolve();
+jest.mock('../../../BaseApi', () => jest.fn());
```

Comment:
I'm not sure I fully understand this (but that's true for all the weird mock stuff :sweat_smile: ). One suggestion I have which seems a bit easier is to remove line 7-14 and instead just add `API.getModule = jest.fn(() => Promise.resolve());`  in the test function below. You can then just do ` await waitFor( ()=> {expect(API.getModule).toHaveBeenCalled();});` instead of the `await act(() => promise)`, what do you think?

Reply:
yes i am also not following the mocking part 100%, the promise way I found from Kent C Dodds( think thats the creator of testing-lilbrary). I didn't like it actually because of the extra `promise` variable.

Codewise I like your approach more. But also technically i am not testing the getModule and therefore testing it `toHaveBeenCalled` seems really correct.

anyways, it is indeed more readable so lets go with it :) Thanks!

Reply:
hmm, good point, I didn't take that into account